src/legacy/core_plugins/console/np_ready/public/legacy.ts

```diff
@@ -64,12 +64,14 @@ uiRoutes.when('/dev_tools/console', {
         throw new Error(message);
       }
 
+      let unmount: AppUnmount | Promise<AppUnmount>;
```

Comment:
Could you clarify for me the scenario in which unmount would be assigned a promise value?

Reply:
This was done to satisfy the type constraints introduced by core: https://github.com/elastic/kibana/blob/master/src/core/public/application/types.ts#L82